src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/CodeActions/CodeActionEndpoint.cs

```diff
@@ -25,6 +26,7 @@ internal class CodeActionEndpoint : ICodeActionHandler
         private readonly ForegroundDispatcher _foregroundDispatcher;
         private readonly DocumentResolver _documentResolver;
         private readonly ILanguageServer _languageServer;
+        private readonly LSPEditorFeatureDetector _lspEditorFeatureDetector;
```

Comment:
I forgot we added this to the workspaces dll. That being said if we're going to be using this type to derive meaning on if VS supports a certain feature we may want to add loads of comments in all the language server areas so future onlookers understand what's going on (even if it's just temporary). The alternative would be to create an interface in our language server with a default implementation, implement it on the VS side and replace its usage.

Reply:
Updated with `LanguageServerFeatureOptions` as discussed.